In [76]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

%matplotlib inline

In [77]:
summer = pd.read_csv('/Users/tolulopeogunfuwa/Desktop/Desktop/2021_Files/DATA ANALYTICS/PRACTISE/HAMOYE/STAGE B/MEDAL/summer.csv')
winter = pd.read_csv('/Users/tolulopeogunfuwa/Desktop/Desktop/2021_Files/DATA ANALYTICS/PRACTISE/HAMOYE/STAGE B/MEDAL/winter.csv')
dictionary = pd.read_csv('/Users/tolulopeogunfuwa/Desktop/Desktop/2021_Files/DATA ANALYTICS/PRACTISE/HAMOYE/STAGE B/MEDAL/dictionary.csv')


In [78]:
summer.head()

Year    City     Sport Discipline             Athlete Country Gender  \
0  1896  Athens  Aquatics   Swimming       HAJOS, Alfred     HUN    Men   
1  1896  Athens  Aquatics   Swimming    HERSCHMANN, Otto     AUT    Men   
2  1896  Athens  Aquatics   Swimming   DRIVAS, Dimitrios     GRE    Men   
3  1896  Athens  Aquatics   Swimming  MALOKINIS, Ioannis     GRE    Men   
4  1896  Athens  Aquatics   Swimming  CHASAPIS, Spiridon     GRE    Men   

                        Event   Medal  
0              100M Freestyle    Gold  
1              100M Freestyle  Silver  
2  100M Freestyle For Sailors  Bronze  
3  100M Freestyle For Sailors    Gold  
4  100M Freestyle For Sailors  Silver

In [79]:
winter.head()

Year      City     Sport Discipline               Athlete Country Gender  \
0  1924  Chamonix  Biathlon   Biathlon           BERTHET, G.     FRA    Men   
1  1924  Chamonix  Biathlon   Biathlon        MANDRILLON, C.     FRA    Men   
2  1924  Chamonix  Biathlon   Biathlon   MANDRILLON, Maurice     FRA    Men   
3  1924  Chamonix  Biathlon   Biathlon       VANDELLE, André     FRA    Men   
4  1924  Chamonix  Biathlon   Biathlon  AUFDENBLATTEN, Adolf     SUI    Men   

             Event   Medal  
0  Military Patrol  Bronze  
1  Military Patrol  Bronze  
2  Military Patrol  Bronze  
3  Military Patrol  Bronze  
4  Military Patrol    Gold

In [80]:
dictionary.head()

Country Code  Population  GDP per Capita
0      Afghanistan  AFG  32526562.0      594.323081
1          Albania  ALB   2889167.0     3945.217582
2          Algeria  ALG  39666519.0     4206.031232
3  American Samoa*  ASA     55538.0             NaN
4          Andorra  AND     70473.0             NaN

In [81]:
dictionary_key = dictionary.drop(['Country'], axis = 1)
dictionary_key.head()

Code  Population  GDP per Capita
0  AFG  32526562.0      594.323081
1  ALB   2889167.0     3945.217582
2  ALG  39666519.0     4206.031232
3  ASA     55538.0             NaN
4  AND     70473.0             NaN

In [82]:
dictionary_keys = dictionary_key.rename(columns={'Code': 'Country'})
dictionary_keys.head()

Country  Population  GDP per Capita
0     AFG  32526562.0      594.323081
1     ALB   2889167.0     3945.217582
2     ALG  39666519.0     4206.031232
3     ASA     55538.0             NaN
4     AND     70473.0             NaN

In [83]:
dictionary_keys.shape

(201, 3)

In [84]:
len(dictionary_keys['Country'].unique().tolist())

201

In [85]:
summer.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31165 entries, 0 to 31164
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Year        31165 non-null  int64 
 1   City        31165 non-null  object
 2   Sport       31165 non-null  object
 3   Discipline  31165 non-null  object
 4   Athlete     31165 non-null  object
 5   Country     31161 non-null  object
 6   Gender      31165 non-null  object
 7   Event       31165 non-null  object
 8   Medal       31165 non-null  object
dtypes: int64(1), object(8)
memory usage: 2.1+ MB


In [86]:
winter.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5770 entries, 0 to 5769
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Year        5770 non-null   int64 
 1   City        5770 non-null   object
 2   Sport       5770 non-null   object
 3   Discipline  5770 non-null   object
 4   Athlete     5770 non-null   object
 5   Country     5770 non-null   object
 6   Gender      5770 non-null   object
 7   Event       5770 non-null   object
 8   Medal       5770 non-null   object
dtypes: int64(1), object(8)
memory usage: 405.8+ KB


In [87]:
dictionary.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 201 entries, 0 to 200
Data columns (total 4 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country         201 non-null    object 
 1   Code            201 non-null    object 
 2   Population      196 non-null    float64
 3   GDP per Capita  176 non-null    float64
dtypes: float64(2), object(2)
memory usage: 6.4+ KB


In [88]:
dictionary_keys.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 201 entries, 0 to 200
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country         201 non-null    object 
 1   Population      196 non-null    float64
 2   GDP per Capita  176 non-null    float64
dtypes: float64(2), object(1)
memory usage: 4.8+ KB


In [89]:
is_NaN = dictionary_keys.isnull()
row_has_NaN = is_NaN.any(axis=1)
rows_with_NaN = dictionary_keys[row_has_NaN]
print(rows_with_NaN)

    Country  Population  GDP per Capita
3       ASA     55538.0             NaN
4       AND     70473.0             NaN
9       ARU    103889.0             NaN
20      BER     65235.0             NaN
27      IVB     30117.0             NaN
36      CAY     59967.0             NaN
45      COK         NaN             NaN
49      CUB  11389562.0             NaN
61      ERI         NaN             NaN
74      GUM    169885.0             NaN
86      IRI  79109272.0             NaN
96      PRK  25155317.0             NaN
105     LBA   6278438.0             NaN
106     LIE     37531.0             NaN
116     MTN   4067564.0             NaN
121     MON     37731.0             NaN
130     AHO         NaN             NaN
139     PLE         NaN             NaN
141     PNG   7619321.0             NaN
147     PUR   3474182.0             NaN
156     SMR     31781.0             NaN
176     SYR  18502413.0             NaN
177     TPE         NaN             NaN
195     VEN  31108083.0             NaN


In [90]:
IS_NaN = dictionary.isnull()
row_Has_NaN = IS_NaN.any(axis=1)
rows_With_NaN = dictionary[row_Has_NaN]
print(rows_With_NaN)

                             Country Code  Population  GDP per Capita
3                    American Samoa*  ASA     55538.0             NaN
4                            Andorra  AND     70473.0             NaN
9                             Aruba*  ARU    103889.0             NaN
20                          Bermuda*  BER     65235.0             NaN
27            British Virgin Islands  IVB     30117.0             NaN
36                   Cayman Islands*  CAY     59967.0             NaN
45                      Cook Islands  COK         NaN             NaN
49                              Cuba  CUB  11389562.0             NaN
61                           Eritrea  ERI         NaN             NaN
74                              Guam  GUM    169885.0             NaN
86                              Iran  IRI  79109272.0             NaN
96                      Korea, North  PRK  25155317.0             NaN
105                            Libya  LBA   6278438.0             NaN
106                 

### Fill null values with zero (0)

In [91]:
dictionary_keys['Population']=dictionary_keys['Population'].fillna(0)
dictionary_keys['GDP per Capita']=dictionary_keys['GDP per Capita'].fillna(0)

In [92]:
is_NaN = dictionary_keys.isnull()
row_has_NaN = is_NaN.any(axis=1)
rows_with_NaN = dictionary_keys[row_has_NaN]
print(rows_with_NaN)

Empty DataFrame
Columns: [Country, Population, GDP per Capita]
Index: []


In [93]:
dictionary_keys.isnull().values.any()

False

### How many Countries have won at least a medal in the Summer Olympics (1896 - 2012)?

In [94]:
len(summer['Country'].unique().tolist())

148

### How many Countries have won at least a medal in the Winter Olympics (1924 - 2014)? 

In [95]:
len(winter['Country'].unique().tolist())

45

In [96]:
len(summer['Medal'].unique().tolist())

3

In [97]:
len(winter['Medal'].unique().tolist())

3

### How many Summer Olympics were hosted between 1896 - 2012

In [98]:
len(summer['Year'].unique().tolist())

27

### How many Winter Olympics were hosted between 1924 - 2014?

In [99]:
len(winter['Year'].unique().tolist())

22

### How many Disciplines were done in the Summer Olympics between 1896 - 2012?

In [100]:
len(summer['Discipline'].unique().tolist())

67

### How many Disciplines were done in the Winter Olympics between 1924 - 2014?

In [101]:
len(winter['Discipline'].unique().tolist())

15

In [102]:
summer.groupby('Country').first()

Year         City       Sport   Discipline                 Athlete  \
Country                                                                       
AFG      2008      Beijing   Taekwondo    Taekwondo        NIKPAI, Rohullah   
AHO      1988        Seoul     Sailing      Sailing         BOERSMA, Jan D.   
ALG      1984  Los Angeles      Boxing       Boxing          ZAOUI, Mohamed   
ANZ      1908       London    Aquatics     Swimming   BEAUREPAIRE, Frank E.   
ARG      1924        Paris   Athletics    Athletics           BRUNETO, Luis   
...       ...          ...         ...          ...                     ...   
VIE      2000       Sydney   Taekwondo    Taekwondo         TRAN, Hieu Ngan   
YUG      1924        Paris  Gymnastics  Artistic G.           STUKELJ, Leon   
ZAM      1984  Los Angeles      Boxing       Boxing            MWILA, Keith   
ZIM      1980       Moscow      Hockey       Hockey  BOXHALL, Arlene Nadine   
ZZX      1896       Athens      Tennis       Tennis            FLACK, Edwin   

        Gender                     Event   Medal  
Country                                           
AFG        Men                   - 58 KG  Bronze  
AHO        Men       Board (Division Ii)  Silver  
ALG        Men                   71-75KG  Bronze  
ANZ        Men           1500M Freestyle  Bronze  
ARG        Men               Triple Jump  Silver  
...        ...                       ...     ...  
VIE      Women                49 - 57 KG  Silver  
YUG        Men            Horizontal Bar    Gold  
ZAM        Men  - 48KG (Light-Flyweight)  Bronze  
ZIM      Women                    Hockey    Gold  
ZZX        Men                   Doubles  Bronze  

[147 rows x 8 columns]

In [103]:
winter.groupby('Country').first()

Year               City       Sport                 Discipline  \
Country                                                                   
AUS      1994        Lillehammer     Skating  Short Track Speed Skating   
AUT      1924           Chamonix     Skating             Figure skating   
BEL      1924           Chamonix   Bobsleigh                  Bobsleigh   
BLR      1994        Lillehammer    Biathlon                   Biathlon   
BUL      1980        Lake Placid      Skiing       Cross Country Skiing   
CAN      1924           Chamonix  Ice Hockey                 Ice Hockey   
CHN      1992        Albertville     Skating  Short Track Speed Skating   
CRO      2002     Salt Lake City      Skiing              Alpine Skiing   
CZE      1998             Nagano  Ice Hockey                 Ice Hockey   
DEN      1998             Nagano     Curling                    Curling   
ESP      1972            Sapporo      Skiing              Alpine Skiing   
EST      2002     Salt Lake City      Skiing       Cross Country Skiing   
EUA      1956  Cortina d'Ampezzo      Skiing              Alpine Skiing   
EUN      1992        Albertville    Biathlon                   Biathlon   
FIN      1924           Chamonix    Biathlon                   Biathlon   
FRA      1924           Chamonix    Biathlon                   Biathlon   
FRG      1952               Oslo   Bobsleigh                  Bobsleigh   
GBR      1924           Chamonix   Bobsleigh                  Bobsleigh   
GDR      1968           Grenoble        Luge                       Luge   
GER      1928          St.Moritz   Bobsleigh                  Bobsleigh   
HUN      1932        Lake Placid     Skating             Figure skating   
ITA      1948          St.Moritz   Bobsleigh                   Skeleton   
JPN      1956  Cortina d'Ampezzo      Skiing              Alpine Skiing   
KAZ      1994        Lillehammer      Skiing       Cross Country Skiing   
KOR      1992        Albertville     Skating  Short Track Speed Skating   
LAT      2006              Turin        Luge                       Luge   
LIE      1976          Innsbruck      Skiing              Alpine Skiing   
LUX      1992        Albertville      Skiing              Alpine Skiing   
NED      1952               Oslo     Skating              Speed skating   
NOR      1924           Chamonix     Skating              Speed skating   
NZL      1992        Albertville      Skiing              Alpine Skiing   
POL      1956  Cortina d'Ampezzo      Skiing            Nordic Combined   
PRK      1964          Innsbruck     Skating              Speed skating   
ROU      1968           Grenoble   Bobsleigh                  Bobsleigh   
RUS      1994        Lillehammer    Biathlon                   Biathlon   
SLO      1994        Lillehammer      Skiing              Alpine Skiing   
SUI      1924           Chamonix    Biathlon                   Biathlon   
SVK      2006              Turin      Skiing                  Snowboard   
SWE      1924           Chamonix     Curling                    Curling   
TCH      1928          St.Moritz      Skiing                Ski Jumping   
UKR      1994        Lillehammer    Biathlon                   Biathlon   
URS      1956  Cortina d'Ampezzo  Ice Hockey                 Ice Hockey   
USA      1924           Chamonix  Ice Hockey                 Ice Hockey   
UZB      1994        Lillehammer      Skiing           Freestyle Skiing   
YUG      1984           Sarajevo      Skiing              Alpine Skiing   

                            Athlete Gender                 Event   Medal  
Country                                                                   
AUS                BRADBURY, Steven    Men           5000M Relay  Bronze  
AUT                    BÖCKL, Willy    Men            Individual  Silver  
BEL                  MORTIAUX, René    Men              Four-Man  Bronze  
BLR            PARAMYGINA, Svetlana  Women                 7.5KM  Silver  
BUL                   LEBANOV, Iv

### Concatenate the Summer & Winter Dataset

In [104]:
summmer_winter_concat = pd.concat([summer, winter], ignore_index=True)

In [105]:
summmer_winter_concat.head()

Year    City     Sport Discipline             Athlete Country Gender  \
0  1896  Athens  Aquatics   Swimming       HAJOS, Alfred     HUN    Men   
1  1896  Athens  Aquatics   Swimming    HERSCHMANN, Otto     AUT    Men   
2  1896  Athens  Aquatics   Swimming   DRIVAS, Dimitrios     GRE    Men   
3  1896  Athens  Aquatics   Swimming  MALOKINIS, Ioannis     GRE    Men   
4  1896  Athens  Aquatics   Swimming  CHASAPIS, Spiridon     GRE    Men   

                        Event   Medal  
0              100M Freestyle    Gold  
1              100M Freestyle  Silver  
2  100M Freestyle For Sailors  Bronze  
3  100M Freestyle For Sailors    Gold  
4  100M Freestyle For Sailors  Silver

In [106]:
summmer_winter_concat.tail()

Year   City   Sport Discipline             Athlete Country Gender  \
36930  2014  Sochi  Skiing  Snowboard        JONES, Jenny     GBR  Women   
36931  2014  Sochi  Skiing  Snowboard     ANDERSON, Jamie     USA  Women   
36932  2014  Sochi  Skiing  Snowboard  MALTAIS, Dominique     CAN  Women   
36933  2014  Sochi  Skiing  Snowboard        SAMKOVA, Eva     CZE  Women   
36934  2014  Sochi  Skiing  Snowboard    TRESPEUCH, Chloe     FRA  Women   

                 Event   Medal  
36930       Slopestyle  Bronze  
36931       Slopestyle    Gold  
36932  Snowboard Cross  Silver  
36933  Snowboard Cross    Gold  
36934  Snowboard Cross  Bronze

In [107]:
summmer_winter_concat.shape

(36935, 9)

In [108]:
summer.shape

(31165, 9)

In [109]:
winter.shape

(5770, 9)

### Total number of Countries that won medals in both the Summer & Winter Olympics (1896 - 2014)

In [110]:
len(summmer_winter_concat['Country'].unique().tolist())

149

### Merge the Summer data with the Dictionary data

In [124]:
summer_merge = pd.merge(summer, dictionary_keys, on='Country')
summer_merge.head()

Year    City      Sport Discipline           Athlete Country Gender  \
0  1896  Athens   Aquatics   Swimming     HAJOS, Alfred     HUN    Men   
1  1896  Athens   Aquatics   Swimming     HAJOS, Alfred     HUN    Men   
2  1896  Athens  Athletics  Athletics  SZOKOLYI, Alajos     HUN    Men   
3  1896  Athens  Athletics  Athletics      DANI, Nandor     HUN    Men   
4  1896  Athens  Athletics  Athletics    KELLNER, Gyula     HUN    Men   

             Event   Medal  Population  GDP per Capita  
0   100M Freestyle    Gold   9844686.0     12363.54346  
1  1200M Freestyle    Gold   9844686.0     12363.54346  
2             100M  Bronze   9844686.0     12363.54346  
3             800M  Silver   9844686.0     12363.54346  
4         Marathon  Bronze   9844686.0     12363.54346

In [125]:
summer_merge.tail()

Year    City      Sport Discipline           Athlete Country Gender  \
25737  2012  London  Athletics  Athletics   BARRONDO, Erick     GUA    Men   
25738  2012  London  Athletics  Athletics     JAMES, Kirani     GRN    Men   
25739  2012  London  Athletics  Athletics       AMOS, Nijel     BOT    Men   
25740  2012  London    Sailing    Sailing  KONTIDES, Pavlos     CYP    Men   
25741  2012  London  Taekwondo  Taekwondo    OBAME, Anthony     GAB    Men   

           Event   Medal  Population  GDP per Capita  
25737  20KM Walk  Silver  16342897.0     3903.478856  
25738       400M    Gold    106825.0     9212.020352  
25739       800M  Silver   2262485.0     6360.138220  
25740      Laser  Silver   1165300.0    23242.840070  
25741    + 80 KG  Silver   1725292.0     8266.445605

In [126]:
len(summer_merge['Country'].unique().tolist())

129

In [127]:
summer_merge.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 25742 entries, 0 to 25741
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Year            25742 non-null  int64  
 1   City            25742 non-null  object 
 2   Sport           25742 non-null  object 
 3   Discipline      25742 non-null  object 
 4   Athlete         25742 non-null  object 
 5   Country         25742 non-null  object 
 6   Gender          25742 non-null  object 
 7   Event           25742 non-null  object 
 8   Medal           25742 non-null  object 
 9   Population      25742 non-null  float64
 10  GDP per Capita  25742 non-null  float64
dtypes: float64(2), int64(1), object(8)
memory usage: 2.4+ MB


### Merge the Winter data with the Dictionary data

In [128]:
winter_merge = pd.merge(winter, dictionary_keys, on='Country')
winter_merge.head()

Year      City     Sport Discipline              Athlete Country Gender  \
0  1924  Chamonix  Biathlon   Biathlon          BERTHET, G.     FRA    Men   
1  1924  Chamonix  Biathlon   Biathlon       MANDRILLON, C.     FRA    Men   
2  1924  Chamonix  Biathlon   Biathlon  MANDRILLON, Maurice     FRA    Men   
3  1924  Chamonix  Biathlon   Biathlon      VANDELLE, André     FRA    Men   
4  1924  Chamonix   Curling    Curling          ALDEERT, H.     FRA    Men   

             Event   Medal  Population  GDP per Capita  
0  Military Patrol  Bronze  66808385.0      36205.5681  
1  Military Patrol  Bronze  66808385.0      36205.5681  
2  Military Patrol  Bronze  66808385.0      36205.5681  
3  Military Patrol  Bronze  66808385.0      36205.5681  
4          Curling  Bronze  66808385.0      36205.5681

In [129]:
winter_merge.tail()

Year       City     Sport Discipline             Athlete Country Gender  \
4821  2006      Turin    Skiing  Snowboard     ZIDEK, Radoslav     SVK    Men   
4822  2010  Vancouver  Biathlon   Biathlon  KUZMINA, Anastazia     SVK  Women   
4823  2010  Vancouver  Biathlon   Biathlon       HURAJT, Pavol     SVK    Men   
4824  2010  Vancouver  Biathlon   Biathlon  KUZMINA, Anastazia     SVK  Women   
4825  2014      Sochi  Biathlon   Biathlon  KUZMINA, Anastazia     SVK  Women   

                Event   Medal  Population  GDP per Capita  
4821  Snowboard Cross  Silver   5424050.0     16088.27759  
4822     10KM Pursuit  Silver   5424050.0     16088.27759  
4823  15KM Mass Start  Bronze   5424050.0     16088.27759  
4824            7.5KM    Gold   5424050.0     16088.27759  
4825            7.5KM    Gold   5424050.0     16088.27759

In [130]:
len(winter_merge['Country'].unique().tolist())

37

In [131]:
winter_merge.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4826 entries, 0 to 4825
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Year            4826 non-null   int64  
 1   City            4826 non-null   object 
 2   Sport           4826 non-null   object 
 3   Discipline      4826 non-null   object 
 4   Athlete         4826 non-null   object 
 5   Country         4826 non-null   object 
 6   Gender          4826 non-null   object 
 7   Event           4826 non-null   object 
 8   Medal           4826 non-null   object 
 9   Population      4826 non-null   float64
 10  GDP per Capita  4826 non-null   float64
dtypes: float64(2), int64(1), object(8)
memory usage: 452.4+ KB


### Concatenate the merged Summer data with the merged Winter data

In [132]:
concat_ = pd.concat([summer_merge, winter_merge], ignore_index=True)

In [133]:
concat_.head()

Year    City      Sport Discipline           Athlete Country Gender  \
0  1896  Athens   Aquatics   Swimming     HAJOS, Alfred     HUN    Men   
1  1896  Athens   Aquatics   Swimming     HAJOS, Alfred     HUN    Men   
2  1896  Athens  Athletics  Athletics  SZOKOLYI, Alajos     HUN    Men   
3  1896  Athens  Athletics  Athletics      DANI, Nandor     HUN    Men   
4  1896  Athens  Athletics  Athletics    KELLNER, Gyula     HUN    Men   

             Event   Medal  Population  GDP per Capita  
0   100M Freestyle    Gold   9844686.0     12363.54346  
1  1200M Freestyle    Gold   9844686.0     12363.54346  
2             100M  Bronze   9844686.0     12363.54346  
3             800M  Silver   9844686.0     12363.54346  
4         Marathon  Bronze   9844686.0     12363.54346

In [134]:
concat_.tail()

Year       City     Sport Discipline             Athlete Country  \
30563  2006      Turin    Skiing  Snowboard     ZIDEK, Radoslav     SVK   
30564  2010  Vancouver  Biathlon   Biathlon  KUZMINA, Anastazia     SVK   
30565  2010  Vancouver  Biathlon   Biathlon       HURAJT, Pavol     SVK   
30566  2010  Vancouver  Biathlon   Biathlon  KUZMINA, Anastazia     SVK   
30567  2014      Sochi  Biathlon   Biathlon  KUZMINA, Anastazia     SVK   

      Gender            Event   Medal  Population  GDP per Capita  
30563    Men  Snowboard Cross  Silver   5424050.0     16088.27759  
30564  Women     10KM Pursuit  Silver   5424050.0     16088.27759  
30565    Men  15KM Mass Start  Bronze   5424050.0     16088.27759  
30566  Women            7.5KM    Gold   5424050.0     16088.27759  
30567  Women            7.5KM    Gold   5424050.0     16088.27759

In [135]:
len(concat_['Country'].unique().tolist())

130

In [136]:
concat_.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30568 entries, 0 to 30567
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Year            30568 non-null  int64  
 1   City            30568 non-null  object 
 2   Sport           30568 non-null  object 
 3   Discipline      30568 non-null  object 
 4   Athlete         30568 non-null  object 
 5   Country         30568 non-null  object 
 6   Gender          30568 non-null  object 
 7   Event           30568 non-null  object 
 8   Medal           30568 non-null  object 
 9   Population      30568 non-null  float64
 10  GDP per Capita  30568 non-null  float64
dtypes: float64(2), int64(1), object(8)
memory usage: 2.6+ MB
